### PR TITLE
Move Attack sort from str to initalization

### PR DESCRIPTION
**Pull request by 1drturtle**
### Summary
Moves attack list sorting from `str` to class initialization. This fixes the issue of attacks being in different spots in the data vs the str.

#### PR Type

- [ ] This PR is a code change that implements a feature request.
- [x] This PR fixes an issue.
- [ ] This PR adds a new feature that is not an open feature request.
- [ ] This PR is not a code change (e.g. documentation, README, ...)
#### Other
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] If code changes were made then they have been tested.
- [ ] I have updated the documentation to reflect the changes.

### CODE REVIEW

**Comment by mommothazaz123** [comment]
I'm not sure this is the right direction to go, since this causes a semantic data change on character commit rather than just changing the display. Additionally, you'd have to do a resort on each modification operation (append, extend, etc).

Why not add the sorting code to `__str__()` as well instead?

**Comment by 1drturtle** [comment]
Alright, so this FR seems to have a lot of issues that I didn't think about, so I'm going to revert the sorting

**Comment by mommothazaz123** [comment]
TL;DR:  After a conversation in Discord we determined that changing attack sorting on the `AttackList` level would cause issues in other `StatBlock` models like monsters.
